pkg/fqdn/namemanager/gc_cache.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package namemanager
+
+import (
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+// DefaultGCCacheSize is the default size of the GC cache
+const DefaultGCCacheSize = 10000
+
+// GCCache tracks recently garbage-collected FQDN entries.
+type GCCache struct {
+	cache *lru.Cache[string, time.Time]
```

Comment:
rather than an LRU cache, what about a ring-buffer of `(timestamp, name, IPs)` entries? That would solve a major problem with this code, namely, we don't have a good idea when a given IP was removed for a given name.